ffmpeg/transcoder.c

```diff
@@ -257,12 +264,17 @@ int transcode(struct transcode_thread *h,
 
       ipkt->pts += ictx->dts_diff;
       ipkt->dts += ictx->dts_diff;
-
-      if (ipkt->stream_index == 0) {
-        ictx->last_dts = ipkt->dts;
-        if (ipkt->duration) {
-          ictx->last_duration = ipkt->duration;
+      if (ipkt->stream_index < MAX_OUTPUT_SIZE) {
+        if (last_dts[ipkt->stream_index] > -1 && ipkt->dts <= last_dts[ipkt->stream_index])  {
+          // skip packet if dts is equal or lesst than previous one
+          goto whileloop_end;
         }
+        last_dts[ipkt->stream_index] = ipkt->dts;
+      }
+
+      ictx->last_dts = ipkt->dts > ictx->last_dts ? ipkt->dts : ictx->last_dts;
```

Comment:
IMO much easier to read:
```suggestion
      if (ictx->last_dts < ipkt->dts) ictx->last_dts = ipkt->dts;
```